eucaconsole/views/vpcs.py

```
@@ -738,3 +743,206 @@ def get_controller_options_json(self):
             ),
         }))
+
+
+class RouteTableMixin(object):
+    def update_routes(self, request, vpc_conn, route_table):
+        routes_json = request.params.get('routes')
+        new_routes = json.loads(routes_json)
+        self.delete_routes(vpc_conn, route_table)
+        self.add_routes(vpc_conn, route_table, new_routes)
+
+    @staticmethod
+    def delete_routes(vpc_conn, route_table):
+        for route in route_table.routes:
+            # Skip deleting local route
+            if route.gateway_id != 'local':
+                vpc_conn.delete_route(route_table.id, route.destination_cidr_block)
+
+    @staticmethod
+    def add_routes(vpc_conn, route_table, new_routes):
+        for route in new_routes:
+            route_target_id = route.get('target')
+            params = dict(
+                route_table_id=route_table.id,
+                destination_cidr_block=route.get('destination_cidr_block')
+            )
+            if route_target_id:
+                if route_target_id.startswith('igw-'):
+                    params.update(dict(gateway_id=route_target_id))
+                elif route_target_id.startswith('eni-'):
+                    params.update(dict(interface_id=route_target_id))
+                # TODO: Handle routes with NAT Gateway target
+                vpc_conn.create_route(**params)
+
+
+class RouteTableView(TaggedItemView, RouteTableMixin):
+    VIEW_TEMPLATE = '../templates/vpcs/route_table_view.pt'
+
+    def __init__(self, request, **kwargs):
+        super(RouteTableView, self).__init__(request, **kwargs)
+        self.location = self.request.route_path(
+            'route_table_view', vpc_id=self.request.matchdict.get('vpc_id'), id=self.request.matchdict.get('id'))
+        self.conn = self.get_connection()
+        self.vpc_conn = self.get_connection(conn_type='vpc')
+        with boto_error_handler(request, self.location):
+            self.vpc = self.get_vpc()
+            self.route_table = self.get_route_table()
```

Actually, I just tested this and the __init__ can raise HTTPNotFound. If this view is only used for existing route tables, that could be cleaner. fails faster, blah, blah.